os_migrate/plugins/module_utils/image.py

```diff
@@ -89,6 +89,14 @@ class Image(resource.Resource):
         'ramdisk_id',
     ]
 
+    # Some parameters are allowed when reading an image resource but not when
+    # creating or updating it.  This list of parameter names is purged from the parameter
+    # list before calling create or update and removed from object properties when reading from sdk
+    # Attribute 'stores' is read-only now. Due to bug https://bugs.launchpad.net/glance/+bug/1889676 glance now
+    # sets stores as a read only property. As such we should remove it from any create or update calls.
+    # An existing pattern is to use the readonly_sdk_params list to do this.
+    readonly_sdk_params = ["store", "stores"]
```

Comment:
The readonly_sdk_params are related to what is directly in the 'params' dictionary (if you look at the serialization in images.yml, or how readonly_sdk_params is used in resource.py). Since we've removed 'store' from 'params' into '_info' above, we no longer need this part.

Re using this to filter 'properties' below on line 183 - i wouldn't double this variable's meaning and use it also as a filter for the 'properties' key in 'params' dictionary. IMO we should have one variable for filtering `['params']` (we do have it inherited from resource.py but no longer need to override it here) and another variable for filtering `['params']['properties']`. I do think it's a good idea to have a variable for this, could be class-wide but maybe having the variable locally in a function that will do the filtering would suffice?

Reply:
Sorry i got things mixed up here. On a second look, the way you used the variable does not mean we're doubling its meaning, as it's generally applied to the final `sdk_params` rather than the resource's `params`, so this was fine.

Still, i think we should probably do the filtering on export as mentioned below due to how idempotency is handled, and in that case we shouldn't use this variable as we won't be filtering the sdk_params anymore.